pyproject.toml

```diff
@@ -152,6 +151,7 @@ distlib = "https://bitbucket.org/pypa/distlib/raw/master/LICENSE.txt"
 #
 
 [tool.ruff]
+target-version = "py38"  # Pin Ruff to Python 3.8
```

Comment:
We should add a note for the reason and aim to eventually remove this in a later PR. (I assume this would want us to switch to built-in container classes instead of from `typing`)

Reply:
Yup! We were just talking 'bout this in https://github.com/pypa/pip/issues/13236.